fix flaky test DSEGOG-456

joshdimanteto opened the pull request:
## Description
Enter a description of the changes here

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
closes DSEGOG-456

joshdimanteto committed:
fix flaky test test

joshdimanteto committed:
Comment out FedId user test and add e2e test UI commands

joshdimanteto committed:
Merge branch 'fix-maintenance-routes' into fix-flaky-tests

joshdimanteto committed:
Remove the FEDID test

louise-davies commented on e2e/real/users.spec.ts:
```diff
@@ -1,5 +1,10 @@
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
+const __dirname = path.dirname(__filename); // get the name of the directory
```
Why is this code added? I can't see it being used in this file...

joshdimanteto replied:
I forgot to remove this when debugging on the CI use the snapshots

louise-davies commented on package.json:
```diff
@@ -60,7 +60,9 @@
     "e2e:serve:api": "yarn build:e2e:api && node ./server/e2e-test-server.js",
     "playwright:test:mocked": "docker run -v $PWD:/test -w=/test $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\") --rm --ipc=host -p 9323:9323 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/test --name-only --json |  ...
-    "playwright:test:real": "docker run -v $PWD:/test -w=/test -e \"USE_REAL_API=true\" $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\") --rm --ipc=host -p 9323:9323 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/tes ...
+    "playwright:test:mocked:ui": "docker run -it -v $PWD:/test -w=/test $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\")  --rm --ipc=host -p 8080:8080 -p 3000:3000 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/test  ...
```
This command can by tidied up as it doesn't need to check for CI (plus we end up specifying `-it` twice)

```suggestion
    "playwright:test:mocked:ui": "docker run -it -v $PWD:/test -w=/test $(node -e \"if(process.platform === 'win32'){console.log('-u pwuser')}else{console.log('-u $(id -u):$(id -g)')}\")  --rm --ipc=host -p 8080:8080 -p 3000:3000 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/test --name-only --json | sed -n 's/^.*:\\(.*\\)\"$/\\1/p') yarn playwright test --ui --ui-host=0.0.0.0 --ui-port=8080",
```

louise-davies commented on package.json:
```diff
@@ -60,7 +60,9 @@
-    "playwright:test:real": "docker run -v $PWD:/test -w=/test -e \"USE_REAL_API=true\" $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\") --rm --ipc=host -p 9323:9323 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/tes ...
+    "playwright:test:mocked:ui": "docker run -it -v $PWD:/test -w=/test $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\")  --rm --ipc=host -p 8080:8080 -p 3000:3000 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/test  ...
+    "playwright:test:real": "docker run -v $PWD:/test -w=/test -e \"USE_REAL_API=true\" $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\")  --rm --ipc=host -p 9323:9323 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/te ...
+    "playwright:test:real:ui": "docker run -it -v $PWD:/test -w=/test -e \"USE_REAL_API=true\" $(node -e \"if(process.env.CI !== 'true'){if(process.platform === 'win32'){console.log('-it -u pwuser')}else{console.log('-it -u $(id -u):$(id -g)')}}else{console.log('-e CI')}\") --rm --ipc=host -p 8080:8080 -p 3000:3000 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn  ...
```
Same with this one

```suggestion
    "playwright:test:real:ui": "docker run -it -v $PWD:/test -w=/test -e \"USE_REAL_API=true\" $(node -e \"if(process.platform === 'win32'){console.log('-u pwuser')}else{console.log('-u $(id -u):$(id -g)')}\") --rm --ipc=host -p 8080:8080 -p 3000:3000 --add-host host.docker.internal:host-gateway mcr.microsoft.com/playwright:v$(yarn info @playwright/test --name-only --json | sed -n 's/^.*:\\(.*\\)\"$/\\1/p') yarn playwright test --ui --ui-host=0.0.0.0 --ui-port=8080",
```

joshdimanteto commented on e2e/real/users.spec.ts:
```diff
@@ -1,5 +1,10 @@
+import { fileURLToPath } from 'url';
+
+const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
+const __dirname = path.dirname(__filename); // get the name of the directory
```
```suggestion
```

joshdimanteto committed:
Apply suggestions from code review

Remove unnecessary code from the UI mode e2e script and the user e2e test file

Co-authored-by: Louise Davies <louise.davies@stfc.ac.uk>
